### PR TITLE
Fix layouts with no filler position

### PR DESCRIPTION
**Pull request by samogot**
we have a such chain:
```
controller - upgrader - storage - pathStart
                               \- filler
```
where each next position was first valid adjacent position
in this cange instead of first valid position, "best" valid position will be taken, where "best" is position, which have the biggest number of valid positions adjacent to it
- By taking best position instead of first for upgrader, we reserve some space for potentially several -
upgraders
- By taking best position instead of first for storage, we increase chances that we will have enough positions for pathStart, roads and filer
- For pathStart, first position is enough

### CODE REVIEW

File: src/prototype_roomPosition.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ RoomPosition.prototype.clearPosition = function(target) {
         return true;
       }
     }
-    this.log('Destroying: ' + structure.structureType);
+    console.log('Destroying: ' + structure.structureType);
```
Comment (TooAngel) [inline src/prototype_roomPosition.js]: Introduce `this.log` would be even cooler.

File: src/prototype_room_init.js [hunk 2]
```diff
@@ -68,42 +68,40 @@ Room.prototype.initSetStorageAndPathStart = function() {
...
-    costMatrix.set(fillerPos.x, fillerPos.y, config.layout.creepAvoid);
-    this.setMemoryCostMatrix(costMatrix);
-    let pathFiller = this.getPath(route, 0, 'pathStart', 'filler', true);
-    for (let pos of pathFiller) {
-      this.increaseCostMatrixValue(costMatrix, pos, config.layout.pathAvoid);
-    }
+  let fillerPos = storagePos.getBestNearPosition();
+  // TODO reset all values if the first doesn't fit
+  this.log('Testing fillerPos' + JSON.stringify(fillerPos));
+  this.deleteMemoryPath('pathStart-filler');
```
Comment (TooAngel) [inline src/prototype_room_init.js]: Can this exist?

File: src/prototype_room_init.js [hunk 3]
```diff
@@ -68,42 +68,40 @@ Room.prototype.initSetStorageAndPathStart = function() {
...
+  this.memory.position.creep.filler = fillerPos;
+  costMatrix.set(fillerPos.x, fillerPos.y, config.layout.creepAvoid);
+  this.setMemoryCostMatrix(costMatrix);
```
Comment (TooAngel) [inline src/prototype_room_init.js]: this could be `increaseCostMatrixValue`, too

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 257/441
    Coefficient: 0.582766439909
    Merging in 4 days 2 hours
    Age 0 days 1 hours

**Pull request merged**